hack/push-chart.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+# Copyright 2024 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+DEST_CHART_DIR=${DEST_CHART_DIR:-bin/}
+
+EXTRA_TAG=${EXTRA_TAG:-$(git branch --show-current)} 
+GIT_TAG=${GIT_TAG:-$(git describe --tags --dirty --always)}
+HELM_CHART_REPO=${HELM_CHART_REPO:-gcr.io/k8s-staging-kueue/charts}
+IMAGE_REPO=${IMAGE_REPO:-gcr.io/k8s-staging-kueue/kueue}
+
+HELM=${HELM:-./bin/helm}
+YQ=${YQ:-./bin/yq}
+
+readonly semver_regex='^v?([0-9]+)(\.[0-9]+)?(\.[0-9]+)?(-([0-9A-Za-z\-]+(\.[0-9A-Za-z\-]+)*))?(\+([0-9A-Za-z\-]+(\.[0-9A-Za-z\-]+)*))?$'
+chart_version=${GIT_TAG}
+
+if [[ ${EXTRA_TAG} =~ ${semver_regex} ]]
+then
+	chart_version=${EXTRA_TAG}
+	# Update the image tag and policy
+	${YQ}  e  ".controllerManager.manager.image.repository = \"${IMAGE_REPO}\" | .controllerManager.manager.image.tag = \"${chart_version}\" | .controllerManager.manager.image.pullPolicy = \"IfNotPresent\"" -i charts/kueue/values.yaml 
+fi
```

Comment:
this should be reverted after the build, otherwise the source tree will remain dirty